spicy/toolchain/src/compiler/validator.cc

```diff
@@ -691,8 +698,10 @@ struct VisitorPost : visitor::PreOrder, hilti::validator::VisitorMixIn {
                     if ( ! e )
                         error(e.error(), n);
                     else {
-                        if ( ! hilti::type::same((*e)->type()->type(), builder()->typeBool()) )
-                            error(fmt("&requires expression must be of type bool, but is of type %s ", *(*e)->type()),
+                        if ( auto result = (*e)->type()->type()->tryAs<hilti::type::Result>();
+                             ! result || ! result->dereferencedType()->type()->isA<hilti::type::Void>() )
+                            error(fmt("&requires expression must be of type bool or result<void>, but is of type %s",
+                                      *(*e)->type()),
```

Comment:
Nit, similar to above.

```suggestion
                                      e.value()->type()->print()),
```

Reply:
Same as above.